### DISCUSSION TITLE
Adding support for additional data formats

### DISCUSSION THREAD

**Discussion by Oli4** [MedVisBonn/eyepy#11] [Ideas]
Currently, eyepy can only parse data stored in Heidelberg formats to EyeVolume objects. Since the intention of eyepy is to provide a high level interface for interaction with OCT data, I would be happy to add support for other formats. Unfortunatelly I have no sample data, hence it would be great if someone with sample data could do the implementation and testing with my assistance. I think it is easiest to base support of other file formats on the file readers implemented in [OCT-Converter](https://github.com/marksgraham/OCT-Converter/) by @marksgraham. The following formats are not supported yet by eyepy but have a file reader in OCT-Converter:

+ .fds (Topcon)
+ .img (Zeiss)
+ .oct (Bioptigen)
+ .OCT (Optovue)
+ .dcm (standard / ZeissDicom)

Making these formats accessible through eyepy is a matter of parsing them to EyeVolume objects through a dedicated import function in `eyepy.io.__init__.py`. The only mandatory data for creating an `EyeVolume` object is the OCT data. So the simplest implementation would look something like that

```python
import eyepy as ep
from oct_converter.readers import READERNAME

def import_vendor_format(path_to_file: Union[str, Path]) -> ep.EyeVolume:
    reader = READERNAME(path_to_file)
    # Create a 3D numpy array from all bscans (n_bscans, bscan_height, bscan_width)
    oct_data = np.stack(reader.read_oct_volume().volume, axis=0)
    ev = ep.EyeVolume(data=oct_data)   
    return ev
```

Of course, when available we also want to provide access to 

+ scaling information / other metadata,
+ associated fundus images,
+ layer / other segmentations.

You can find examples for all these cases in the existing import functions in `eyepy.io.__init__.py`

If you plan to work on an implementation let me know, I am happy to help and answer questions. Currently, the following implementations are planned/ongoing:

- .img (Zeiss) by @marksgraham 

## Thank you for your contribution!

- .fda (Topcon) import function implemented by @drombas

**Comment by marksgraham** [comment]
Hi @Oli4 

I can start with Zeiss .img as it's easy, and then work on some more when I get a bit more familiar with your library. It's possible it will take me a bit of time before I can get round to it due to some work commitments atm.

**Reply by Oli4** [reply]
Great! Let me know if any questions arise.

**Comment by Himanshu1331** [comment]
Hi

I have Retinal OCT images which have **.ex.dcm** format, I have tried loading it using eyepie, but I am getting some errors. Can you help me with loading the images?
Regards

**Comment by Oli4** [comment]
Hey, 

do you know the device your data comes from? You might be able to use pydicom to read it. eyepy currently does not support this format but I am always willing to add new formats. If you figure out how to extract the B-scans from your data, let me know and we can add an import function for eyepy. 

Best 
Olivier
